\documentclass[openany]{book}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Supplementary Information for: Hyoliths with pedicles constrain the origin of the brachiopod body plan},
            pdfauthor={Haijing Sun, Martin R. Smith, Han Zeng, Fangchen Zhao, Guoxiang Li and Maoyan Zhu},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{apalike-doi}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}
  \title{Supplementary Information for: \newline\newline Hyoliths with pedicles
constrain the origin of the brachiopod body plan}
  \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
  \author{Haijing Sun, Martin R. Smith, Han Zeng, Fangchen Zhao, Guoxiang Li and
Maoyan Zhu}
  \preauthor{\centering\large\emph}
  \postauthor{\par}
  \predate{\centering\large\emph}
  \postdate{\par}
  \date{2018-05-25}

\usepackage{doi} % Adds hyperlinks to dois
\setcitestyle{round}
\usepackage[nottoc]{tocbibind} % list references in TOC
\raggedbottom % already in pnas-new
\usepackage[section]{placeins}

\usepackage{amsthm}
\newtheorem{theorem}{Theorem}[chapter]
\newtheorem{lemma}{Lemma}[chapter]
\newtheorem{corollary}{Corollary}[chapter]
\newtheorem{proposition}{Proposition}[chapter]
\newtheorem{conjecture}{Conjecture}[chapter]
\theoremstyle{definition}
\newtheorem{definition}{Definition}[chapter]
\theoremstyle{definition}
\newtheorem{example}{Example}[chapter]
\theoremstyle{definition}
\newtheorem{exercise}{Exercise}[chapter]
\theoremstyle{remark}
\newtheorem*{remark}{Remark}
\newtheorem*{solution}{Solution}
\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\chapter*{Supplementary Text}\label{supplementary-text}
\addcontentsline{toc}{chapter}{Supplementary Text}

This document comtains supplementary material to
\citet{Sun2018Hyolithswith}. It is best viewed in HTML format at
\href{https://ms609.github.io/hyoliths/}{ms609.github.io/hyoliths}.

It opens with a detailed discussion of
\protect\hyperlink{treesearch}{analyses} of the
\protect\hyperlink{dataset}{morphological dataset} constructed to
accompany \citet{Sun2018Hyolithswith}, and their results.

The results presented in the main paper employ the algorithm described
by \citet{Brazeau2018} for correct handling of inapplicable data in a
parsimony setting. This document depicts how each character is most
parsimoniously \protect\hyperlink{reconstructions}{reconstructed} on an
optimal tree.

For completeness, we also document the results of
\protect\hyperlink{fitch}{standard Fitch parsimony} analysis, and the
results of \protect\hyperlink{bayesian}{Bayesian analysis}, neither of
which treat inapplicable data in a logically consistent fashion.

Supplementary \protect\hyperlink{figures}{figures} and
\protect\hyperlink{table}{tables} appear after the text.

\hypertarget{dataset}{\chapter{Phylogenetic dataset}\label{dataset}}

Analysis was performed on a new matrix of 37 early brachiozoan taxa,
including hyoliths, tommotiids and mickwitziids, which were coded for
107 morphological characters (62 neomorphic, 45 transformational).

\emph{Namacalathus} was incorporated as a 38\textsuperscript{th} taxon,
but preliminary results did not uphold the homology of its potentially
brachiozoan-like features. As such, we excluded it from our analysis due
to its morphological distance from ingroup taxa, a likely source of long
branch error. \emph{Dailyatia} was instead selected as an outgroup as
camenellans have been interpreted as the earliest diverging members of
the Brachiozoa \citep{Skovsted2015Theearly, Zhao2017}.

Characters are coded following the recommendations of
\citet{Brazeau2018}:

\begin{itemize}
\item
  We have employed reductive coding, using a distinct state to mark
  character inapplicability. Character specifications follow the
  structural syntax of \citet{Sereno2007} in order to highlight
  ontological dependence between characters and emphasize the structure
  of the dataset.
\item
  We have distinguished between neomorphic and transformational
  characters \citep[sensu][]{Sereno2007} by reserving the token
  \texttt{0} to refer to the absence of a neomorphic
  (i.e.~presence/absence) character. The states of transformational
  characters (i.e.~characters that describe a property of a feature) are
  represented by the tokens \texttt{1}, \texttt{2}, \texttt{3}, \ldots{}
\item
  We code the absence of neomorphic ontologically dependent characters
  \citep[sensu][]{Vogt2017} as absence, rather than inapplicability.
\end{itemize}

The complete dataset comprises 3959 character codings, of which 451 are
inapplicable and 2319 were neither ambiguous nor inapplicable. The
amount and quality of data that \emph{is} coded is more instructive than
a measure of how many cells are ambiguous \citep{Wiens1998, Wiens2003}.
Of the 107 characters, the number that were coded with an applicable
token for each taxon is:

\begin{tabular}{l|l|l|l|l|l}
\hline
 &  &  &  &  & \\
\hline
\_Acanthotretella spinosa\_ & 59   \&nbsp; & \_Haplophrentis carinatus\_ & 65   \&nbsp; & \_Orthis\_ & 62   \&nbsp;\\
\hline
\_Alisina\_ & 75   \&nbsp; & \_Heliomedusa orienta\_ & 57   \&nbsp; & \_Paterimitra\_ & 55   \&nbsp;\\
\hline
\_Askepasma toddense\_ & 65   \&nbsp; & \_Kutorgina chengjiangensis\_ & 76   \&nbsp; & \_Pedunculotheca diania\_ & 60   \&nbsp;\\
\hline
\_Antigonambonites planus\_ & 73   \&nbsp; & \_Lingula\_ & 93   \&nbsp; & \_Pelagodiscus atlanticus\_ & 95   \&nbsp;\\
\hline
\_Botsfordia\_ & 64   \&nbsp; & \_Lingulosacculus\_ & 49   \&nbsp; & \_Phoronis\_ & 43   \&nbsp;\\
\hline
\_Clupeafumosus socialis\_ & 66   \&nbsp; & \_Lingulellotreta malongensis\_ & 70   \&nbsp; & \_Salanygolina\_ & 67   \&nbsp;\\
\hline
\_Coolinia pecten\_ & 69   \&nbsp; & \_Longtancunella chengjiangensis\_ & 49   \&nbsp; & \_Siphonobolus priscus\_ & 62   \&nbsp;\\
\hline
\_Craniops\_ & 57   \&nbsp; & \_Micrina\_ & 59   \&nbsp; & \_Terebratulina\_ & 96   \&nbsp;\\
\hline
\_Dailyatia\_ & 32   \&nbsp; & \_Micromitra\_ & 69   \&nbsp; & \_Ussunia\_ & 42   \&nbsp;\\
\hline
\_Eccentrotheca\_ & 31   \&nbsp; & \_Mickwitzia muralensis\_ & 63   \&nbsp; & \_Tomteluva perturbata\_ & 51   \&nbsp;\\
\hline
\_Eoobolus\_ & 70   \&nbsp; & \_Mummpikia nuda\_ & 45   \&nbsp; & \_Yuganotheca elegans\_ & 45   \&nbsp;\\
\hline
\_Glyptoria\_ & 65   \&nbsp; & \_Nisusia sulcata\_ & 73   \&nbsp; &  & \\
\hline
\_Gasconsia\_ & 61   \&nbsp; & \_Novocrania\_ & 86   \&nbsp; &  & \\
\hline
\end{tabular}

The matrix can be viewed interactively and downloaded at Morphobank
(\href{https://morphobank.org/permalink/?P2800}{project 2800}). {[}This
link will become live on publication of the paper. Referees should
follow the pre-publication link to the dataset that has been provided in
the main manuscript.{]}

A static version of the NEXUS file used to generate this supplementary
information can be downloaded directly from
\url{https://raw.githubusercontent.com/ms609/hyoliths/master/mbank_X24932_4-18-2018_656.nex}
.

\hypertarget{treesearch}{\chapter{Parsimony analysis}\label{treesearch}}

The phylogenetic dataset contains a considerable proportion of
inapplicable codings (451/3959 = 11.4\% of tokens), which are known to
introduce error and bias to phylogenetic reconstruction when the Fitch
algorithm is employed \citep{Maddison1993, Brazeau2018}. As such, we
employed a new tree-scoring algorithm that correctly handles
inapplicable data \citep{Brazeau2018}, implemented in the
\emph{MorphyLib} C library \citep{Brazeau2017Morphylib}. We employed the
R package \emph{TreeSearch} v0.1.2 \citep{Smith2018TreeSearch} to
conduct phylogenetic tree search with this algorithm.

As this is a new method, we also employed the traditional, Fitch
algorithm, even though this approach is known to generate erroneous
trees. The results of this analysis can be viewed in
\protect\hyperlink{fitch}{a later section}.

\section{Search parameters}\label{search-parameters}

Heuristic searches were conducted using the parsimony ratchet
\citep{Nixon1999} under equal and implied weights \citep{Goloboff1997}.
The consensus tree presented in the main manuscript represents a strict
consensus of all trees that are most parsimonious under one or more of
the concavity constants (\emph{k}) 2, 3, 4.5, 7, 10.5, 16 and 24, an
approach that has been shown to produce higher accuracy (i.e.~more nodes
and quartets resolved correctly) than equal weights at any given level
of precision \citep{Smith2017}.

\section{Analysis}\label{analysis}

The R commands used to conduct the analysis are reproduced below. The
results can most readily be replicated using the
\href{https://github.com/ms609/hyoliths/}{R markdown files} (.Rmd) used
to generate these pages.

\subsection{Initialize and load data}\label{initialize-and-load-data}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# Load data from locally downloaded copy of MorphoBank matrix}
\NormalTok{my_data <-}\StringTok{ }\KeywordTok{ReadAsPhyDat}\NormalTok{(filename)}
\NormalTok{my_data}\OperatorTok{$}\NormalTok{Namacalathus <-}\StringTok{ }\OtherTok{NULL} \CommentTok{# Exclude Namacalathus}
\NormalTok{iw_data <-}\StringTok{ }\KeywordTok{PrepareDataIW}\NormalTok{(my_data)}
\end{Highlighting}
\end{Shaded}

\subsection{Generate starting tree}\label{generate-starting-tree}

Start by quickly rearranging a neighbour-joining tree, rooted on the
outgroup.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{nj.tree <-}\StringTok{ }\KeywordTok{NJTree}\NormalTok{(my_data)}
\NormalTok{rooted.tree <-}\StringTok{ }\KeywordTok{EnforceOutgroup}\NormalTok{(nj.tree, outgroup)}
\NormalTok{start.tree <-}\StringTok{ }\KeywordTok{TreeSearch}\NormalTok{(}\DataTypeTok{tree=}\NormalTok{rooted.tree, }\DataTypeTok{dataset=}\NormalTok{my_data, }\DataTypeTok{maxIter=}\DecValTok{3000}\NormalTok{,}
                         \DataTypeTok{EdgeSwapper=}\NormalTok{RootedNNISwap, }\DataTypeTok{verbosity=}\DecValTok{0}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Implied weights analysis}\label{implied-weights-analysis}

The position of the root does not affect tree score, so we keep it fixed
(using \texttt{RootedXXXSwap} functions) to avoid unnecessary swaps.

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (k }\ControlFlowTok{in}\NormalTok{ kValues) \{}
\NormalTok{  iw.tree <-}\StringTok{ }\KeywordTok{IWRatchet}\NormalTok{(start.tree, iw_data, }\DataTypeTok{concavity=}\NormalTok{k,}
                       \DataTypeTok{ratchHits =} \DecValTok{60}\NormalTok{, }\DataTypeTok{searchHits=}\DecValTok{55}\NormalTok{,}
                       \DataTypeTok{swappers=}\KeywordTok{list}\NormalTok{(RootedTBRSwap, RootedSPRSwap, RootedNNISwap),}
                       \DataTypeTok{verbosity=}\NormalTok{0L)}
\NormalTok{  score <-}\StringTok{ }\KeywordTok{IWScore}\NormalTok{(iw.tree, iw_data, }\DataTypeTok{concavity=}\NormalTok{k)}
  \CommentTok{# Write a single best tree}
  \KeywordTok{write.nexus}\NormalTok{(iw.tree,}
              \DataTypeTok{file=}\KeywordTok{paste0}\NormalTok{(}\StringTok{"TreeSearch/hy_iw_k"}\NormalTok{, k, }\StringTok{"_"}\NormalTok{, }
                          \KeywordTok{signif}\NormalTok{(score, }\DecValTok{5}\NormalTok{), }\StringTok{".nex"}\NormalTok{, }\DataTypeTok{collapse=}\StringTok{''}\NormalTok{))}

\NormalTok{  iw.consensus <-}\StringTok{ }\KeywordTok{IWRatchetConsensus}\NormalTok{(iw.tree, iw_data, }\DataTypeTok{concavity=}\NormalTok{k,}
                  \DataTypeTok{swappers=}\KeywordTok{list}\NormalTok{(RootedTBRSwap, RootedNNISwap),}
                  \DataTypeTok{searchHits=}\DecValTok{55}\NormalTok{,}
                  \DataTypeTok{nSearch=}\DecValTok{150}\NormalTok{, }\DataTypeTok{verbosity=}\NormalTok{0L)}
  \KeywordTok{write.nexus}\NormalTok{(iw.consensus, }
              \DataTypeTok{file=}\KeywordTok{paste0}\NormalTok{(}\StringTok{"TreeSearch/hy_iw_k"}\NormalTok{, k, }\StringTok{"_"}\NormalTok{, }
                          \KeywordTok{signif}\NormalTok{(}\KeywordTok{IWScore}\NormalTok{(iw.tree, iw_data, }\DataTypeTok{concavity=}\NormalTok{k), }\DecValTok{5}\NormalTok{),}
                          \StringTok{".all.nex"}\NormalTok{, }\DataTypeTok{collapse=}\StringTok{''}\NormalTok{))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\subsection{Equal weights analysis}\label{equal-weights-analysis}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{ew.tree <-}\StringTok{ }\KeywordTok{Ratchet}\NormalTok{(start.tree, my_data, }\DataTypeTok{verbosity=}\NormalTok{0L,}
                   \DataTypeTok{ratchHits =} \DecValTok{25}\NormalTok{, }\DataTypeTok{searchHits=}\DecValTok{55}\NormalTok{, }\CommentTok{# ratchHits = 10 not enough}
                   \DataTypeTok{swappers=}\KeywordTok{list}\NormalTok{(RootedTBRSwap, RootedSPRSwap, RootedNNISwap))}
\NormalTok{ew.consensus <-}\StringTok{ }\KeywordTok{RatchetConsensus}\NormalTok{(ew.tree, my_data, }\DataTypeTok{nSearch=}\DecValTok{150}\NormalTok{, }\DataTypeTok{searchHits =} \DecValTok{55}\NormalTok{,}
                                 \DataTypeTok{swappers=}\KeywordTok{list}\NormalTok{(RootedTBRSwap, RootedNNISwap),}
                                 \DataTypeTok{verbosity=}\NormalTok{0L)}
\KeywordTok{write.nexus}\NormalTok{(ew.consensus, }\DataTypeTok{file=}\KeywordTok{paste0}\NormalTok{(}\DataTypeTok{collapse=}\StringTok{''}\NormalTok{, }\StringTok{"TreeSearch/hy_ew_"}\NormalTok{,}
                                      \KeywordTok{Fitch}\NormalTok{(ew.tree, my_data), }\StringTok{".nex"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\section{Results}\label{results}






\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-maj-consensus-1.pdf}
\caption{\label{fig:treesearch-maj-consensus}Consensus of all parsimony results.
Node labels denote the proportion of trees obtained
under all analytical conditions that support the clade.}
\end{figure}




\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-maj-consensus-pruned-1.pdf}
\caption{\label{fig:treesearch-maj-consensus-pruned}Consensus of all parsimony results.
Node labels denote the proportion of trees obtained
under all analytical conditions that support the clade.}
\end{figure}





\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-iw-consensus-1.pdf}
\caption{\label{fig:treesearch-iw-consensus}Consensus of implied weights analyses at all values of
\emph{k}. Wildcard taxa have been excluded from the consensus tree shown
above to improve resolution.}
\end{figure}








\clearpage 

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-all-iw-results-1.pdf}
\caption{\label{fig:treesearch-all-iw-results}Consensus trees of implied weights analyses
at all values of \emph{k}, and at the individual
values \emph{k} = 2, 3 and 4.5.}
\end{figure}

\clearpage 

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-iw-results-3-1.pdf}
\caption{\label{fig:treesearch-iw-results-3}Consensus trees of implied weights analyses
at \emph{k} = 7, 10.5, 16 and 24.}
\end{figure}

\clearpage

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/treesearch-equal-weights-results-1.pdf}
\caption{\label{fig:treesearch-equal-weights-results}Strict consensus of
most parsimonious trees under equally weighted parsimony}
\end{figure}

\clearpage

\hypertarget{reconstructions}{\chapter{Character
reconstructions}\label{reconstructions}}

This page provides definitions for each of the characters in our matrix,
and justifies codings in particular taxa where relevant. Further
citations for codings that are not discussed in the text can be viewed
by browsing the \protect\hyperlink{dataset}{morphological dataset} on
MorphoBank (\href{https://morphobank.org/permalink/?P2800}{project
2800}). This link will become live on publication of the paper. Referees
should follow the pre-publication link to the dataset that has been
provided in the main manuscript.

Alongside each character's definition, each character is mapped onto a
tree. Here, we have arbitrarily selected one most parsimonious tree
obtained under implied weighting, \(k = 4.5\). Other trees can be viewed
in the HTML version of this file at
\href{https://ms609.github.io/hyoliths/reconstructions.html}{ms609.github.io/hyoliths}.
Each tip is labelled as it is coded in the matrix, and these states are
used to reconstruct the condition of each internal node, using the
parsimony method of \citet{Brazeau2018} as implemented in the
\emph{Inapp} \emph{R} package.

We emphasize that different trees will give different reconstructions.
The character mappings are not intended to definitively establish how
each character evolved, but to help the reader quickly establish how
each character has been coded, and to visualize at a glance how well the
character fits onto the given tree. We consider this more intuitive than
the use of the flawed \citep{Archie1989} Consistency Index, but include
this value because of its historic significance.

\section{Sclerites}\label{sclerites}

\subsection*{{[}1{]} Present in adult}\label{present-in-adult}
\addcontentsline{toc}{subsection}{{[}1{]} Present in adult}

\includegraphics{Brachiopod_phylogeny_files/figure-latex/character-mapping-1.pdf}

\textbf{Character 1: Sclerites: Present in adult}

\begin{quote}
0: Absent\\
1: Present\\
Neomorphic character.
\end{quote}

Plate-like (wider than tall) skeletal elements, whether mineralized or
non-mineralized.\\
The definition deliberately excludes setae (which are taller than wide).

\section{Sclerites: Bivalved {[}2{]}}\label{sclerites-bivalved-2}

\includegraphics{Brachiopod_phylogeny_files/figure-latex/character-mapping-2.pdf}

\textbf{Character 2: Sclerites: Bivalved}

\begin{quote}
0: Scleritomous: without differentiated dorsal and ventral valves\\
1: Bivalved: scleritome dominated by prominent dorsal and ventral
valve\\
Neomorphic character.
\end{quote}

Scleritome dominated by prominent differentiated dorsal and ventral
valves.

\subsection*{{[}3{]} Accessory sclerites
reduced}\label{accessory-sclerites-reduced}
\addcontentsline{toc}{subsection}{{[}3{]} Accessory sclerites reduced}

\includegraphics{Brachiopod_phylogeny_files/figure-latex/character-mapping-3.pdf}

\textbf{Character 3: Sclerites: Bivalved: Accessory sclerites reduced}

\begin{quote}
0: Accessory sclerites present\\
1: Accessory sclerites absent: two valves only\\
Neomorphic character.
\end{quote}

Taxa in the bivalved condition may retain sclerites as small additional
elements, such as the L-elements of \emph{Paterimitra}
\citep{Skovsted2015Theearly}.

This character is treated as neomorphic, with accessory sclerites
ancestrally present, recognizing the likely origin of brachiozoans (and
Lophotrochozoans more generally) from a scleritomous organism.

Coded as inapplicable in taxa that lack multiple skeletal elements.

\hypertarget{Haplophrentis_carinatus-coding-3}{}
\emph{Haplophrentis carinatus}: Coded as ambiguous to recognize the
possibility that helens may correspond to L-elements of
\emph{Paterimitra} \citep{Moysiuk2017Hyolithsare}.

\hypertarget{Paterimitra-coding-3}{}
\emph{Paterimitra}: L-sclerites \citep{Skovsted2009Thescleritome}.

\subsection*{{[}4{]} Hinge line shape}\label{hinge-line-shape}
\addcontentsline{toc}{subsection}{{[}4{]} Hinge line shape}

\includegraphics{Brachiopod_phylogeny_files/figure-latex/character-mapping-4.pdf}

\textbf{Character 4: Sclerites: Bivalved: Hinge line shape}

\begin{quote}
1: Astrophic\\
2: Strophic\\
Transformational character.
\end{quote}

\hypertarget{Botsfordia-coding-4}{}
\emph{Botsfordia}: Coded as dissociated in Williams \emph{et al}.
\citeyearpar{Williams1998Thediversity}, appendix 2.

\hypertarget{Craniops-coding-4}{}
\emph{Craniops}: Astrophic: rounded posterior margin \citep[see fig. 91
in][]{Williams2000LinguliformeaCraniiformea}.

\hypertarget{Gasconsia-coding-4}{}
\emph{Gasconsia}: The straight posterior margin of \emph{Gasconsia}
contributes to an overall resemblance with the Chileids
\citep{Holmer2014Ordovician96}.

\hypertarget{Kutorgina_chengjiangensis-coding-4}{}
\emph{Kutorgina chengjiangensis}: Williams \emph{et al}.
\citeyearpar[p.~208]{Williams2000LinguliformeaCraniiformea} consider the
hinge of \emph{Kutorgina} to be stropic, whereas Bassett \emph{et al}.
\citeyearpar{Bassett2001Functionalmorphology} argue for an astropic
interpretation -- whilst noting that the arrangement is prominently
different from other astrophic taxa. We therefore code this taxon as
ambiguous.

\hypertarget{Longtancunella_chengjiangensis-coding-4}{}
\emph{Longtancunella chengjiangensis}: ``\emph{Longtancunella} has an
oval to subcircular shell with a very short strophic hinge line'' --
\citet{Zhang2011Theexceptionally}.

\hypertarget{Mickwitzia_muralensis-coding-4}{}
\emph{Mickwitzia muralensis}: non-strophic.

\hypertarget{Micrina-coding-4}{}
\emph{Micrina}: See \citet{Holmer2008TheEarly}.

\hypertarget{Nisusia_sulcata-coding-4}{}
\emph{Nisusia sulcata}: ``The strophic, articulated shells of the
Kutorginata rotated on simple hinge mechanisms that are different from
those of other rhynchonelliforms''
\citep[p.~208]{Williams2000LinguliformeaCraniiformea}.

\hypertarget{Novocrania-coding-4}{}
\emph{Novocrania}: Craniides have a strophic posterior valve edge
\citep[table 39 on p.~2853]{Williams2007Supplement}: \emph{Novocrania}'s
``dorsal posterior margin'' is ``straight''
\citep[p.~171]{Williams2000LinguliformeaCraniiformea}.

\hypertarget{Tomteluva_perturbata-coding-4}{}
\emph{Tomteluva perturbata}: ``Tomteluvid taxa all have a strongly
ventribiconvex, astrophic shell with a unisulcate commissure'' --
\citet{Streng2016Anew}, p5.

\hypertarget{Yuganotheca_elegans-coding-4}{}
\emph{Yuganotheca elegans}: Not evident from fossil material; the
possibility of a short strophic hinge line (as in \emph{Longtancunella})
is difficult to discount.

\hypertarget{fitch}{\chapter{Fitch parsimony}\label{fitch}}

Parsimony search was conducted in TNT v1.5 \citep{Goloboff2016} using
ratchet and tree drifting heuristics \citep{Goloboff1999, Nixon1999},
repeating the search until the optimal score had been hit by 1500
independent searches:

\begin{quote}
xmult:rat10 drift10 hits 1500 level 4 chklevel 5;
\end{quote}

Searches were conducted under equal weights and results saved to file:

\begin{quote}
piwe-; xmult; {/* Conduct search with equal weighting */}

tsav *TNT/ew.tre;sav;tsav/; {/* Save results to file */}

keep 0; hold 10000; {/* Clear trees from memory */}
\end{quote}

Further searches were conducted under extended implied weighting
\citep{Goloboff1997, Goloboff2014}, under the concavity constants 2, 3,
4.5, 7, 10.5, 16 and 24:

\begin{quote}
xpiwe=; {/* Enable extended implied weighting */}

piwe=2; xmult; {/* Conduct analysis at k = 2 */}

tsav *TNT/xpiwe2.tre; sav; tsav/; {/* Save results to file */}

keep 0; hold 10000; {/* Clear trees from memory */}

piwe=3; xmult; {/* Conduct analysis at k = 3 */}

tsav *TNT/xpiwe3.tre; sav ;tsav/; {/* Save results to file */}
\end{quote}

We acknowledge the Willi Hennig Society for their sponsorship of the TNT
software.

\section{Results}\label{results-1}









\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/tnt-iw-consensus-1.pdf}
\caption{\label{fig:tnt-iw-consensus}Strict consensus of all trees recovered by TNT
using Fitch parsimony with implied weighting at all values of \emph{k}, and at the individual
values \emph{k} = 2, 3 and 4.5.
The consensus of all implied weights runs is
not very well resolved, largely due to a few wildcard taxa, particularly
at \(k = 4.5\), which obscures a consistent set of relationships between
the remaining taxa.}
\end{figure}

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/tnt-iw-7-24-1.pdf}
\caption{\label{fig:tnt-iw-7-24}Strict consensus of all trees recovered by TNT
using Fitch parsimony with implied weighting, at \emph{k} = 7, 10.5, 16 and 24.}
\end{figure}

\newpage










\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/tnt-ew-consensus-1.pdf}
\caption{\label{fig:tnt-ew-consensus}Consensus of all trees obtained using equally weighted
Fitch parsimony in TNT. \emph{Mickwitzia} and \emph{Micrina} may equally
parsimoniously be reconstructed in the basal region of the linguliform
or rhynchonelliform lineages; as such, the inclusion of these taxa in
the consensus tree reduces resolution. These taxa were still included in
the analysis used to generate this tree, but were removed from each MPT
before the consensus was calculated in order that the relationships that
are present in each tree might be more easily observed.}
\end{figure}

\hypertarget{bayesian}{\chapter{Bayesian analysis}\label{bayesian}}

Bayesian search was conducted in MrBayes v3.2.6 \citep{Ronquist2012}
using the Mk model \citep{Lewis2001} with gamma-distributed rate
variation across characters:

\begin{quote}
lset coding=variable rates=gamma;
\end{quote}

Branch length was drawn from a dirichlet prior distribution, which is
less informative than an exponential model \citep{Rannala2012}, but
requires a prior mean tree length within about two orders of magnitude
of the true value \citep{Zhang2012}. To satisfy this latter criterion,
we specified the prior mean tree length to be equal to the length of the
most parsimonious tree under equal weights, using a Dirichlet prior with
\(\alpha_T = 1\), \(\beta_T = 1/(\)\emph{equal weights tree
length}\(/\)\emph{number of characters}\()\), \(\alpha = c = 1\):

\begin{quote}
prset brlenspr = unconstrained: gammadir(1, 0.34, 1, 1);
\end{quote}

Neomorphic and transformational characters
\citep[\emph{sensu}][]{Sereno2007} were allocated to two separate
partitions whose proportion of invariant characters and gamma shape
parameters were allowed to vary independently:

\begin{quote}
charset Neomorphic = 1 2 3 5 7 8 9 10 12 14 17 18 19 22 23 24 25 28 29
30 31 32 35 37 39 46 47 49 50 51 52 53 54 57 61 62 63 64 66 67 69 70 71
72 73 77 78 79 81 83 86 87 91 93 94 95 97 98 99 102 103 105;

charset Transformational = 4 6 11 13 15 16 20 21 26 27 33 34 36 38 40 41
42 43 44 45 48 55 56 58 59 60 65 68 74 75 76 80 82 84 85 88 89 90 92 96
100 101 104 106 107;

partition chartype = 2: Neomorphic, Transformational;

set partition = chartype;

unlink shape=(all) pinvar=(all);
\end{quote}

Neomorphic characters were not assumed to have a symmetrical transition
rate -- that is, the probability of the absent → present transition was
allowed to differ from that of the present → absent transition, being
drawn from a uniform prior:

\begin{quote}
prset applyto=(1) symdirihyperpr=fixed(1.0);
\end{quote}

The rate of variation in neomorphic characters was also allowed to vary
from that of transformational characters:

\begin{quote}
prset applyto=(1) ratepr=variable;
\end{quote}

\emph{Dailyatia} was selected as an outgroup:

\begin{quote}
outgroup Dailyatia;
\end{quote}

Four MrBayes runs were executed, each sampling eight chains for
5~000~000 generations, with samples taken every 500 generations. The
first 10\% of samples were discarded as burn-in.

\begin{quote}
mcmcp ngen=5000000 samplefreq=500 nruns=4 nchains=8 burninfrac=0.1;
\end{quote}

A posterior tree topology was derived from the combined posterior sample
of all runs. Convergence was indicated by PSRF = 1.00 and an estimated
sample size of \textgreater{} 200 for each parameter.

\section{Parameter estimates}\label{parameter-estimates}

\begin{tabular}{l|r|r|r|r|r}
\hline
Parameter & Mean & Variance & minESS & avgESS & PSRF\\
\hline
TL\{all\} & 7.380 & 0.54000 & 3160 & 5380 & 0.99996\\
\hline
m\{1\} & 0.609 & 0.00984 & 733 & 2790 & 1.00000\\
\hline
\end{tabular}

\section{Results}\label{results-2}

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/mrbayes-full-consensus-1.pdf}
\caption{\label{fig:mrbayes-full-consensus}Results of Bayesian analysis,
posterior probability \textgreater{} 50\%, all taxa}
\end{figure}

\begin{figure}
\centering
\includegraphics{Brachiopod_phylogeny_files/figure-latex/mrbayes-pruned-consensus-1.pdf}
\caption{\label{fig:mrbayes-pruned-consensus}Results of Bayesian analysis,
posterior probability \textgreater{} 50\%, wildcard taxa pruned}
\end{figure}

\chapter{Taxonomic implications}\label{taxonomic-implications}

This section briefly places key features of our results in the context
of previous phylogenetic hypotheses.

\begin{description}
\item[Brachiopod crown and stem group]
Crown- and stem-group terminology has great value in clarifying the
early evolution of major lineages \citep{Budd2000, Carlson2009}. The
crown group of a lineage is defined as the last common ancestor of all
living members of a group, and all its descendants; the stem group as
all taxa more closely related to the crown group than to any other
extant taxon. In our analyses, the brachiopod crown group corresponds to
the last common ancestor of \emph{Terebratulina} and \emph{Lingula}; the
brachiopod stem group comprises anything between this node and the
branching point of \emph{Phoronis}.
\item[Craniiforms]
Trimerellids are reconstructed as paraphyletic with respect to
Craniiforms. This is consistent with the affinity commonly drawn between
these groups \citep[e.g.][]{Williams2000LinguliformeaCraniiformea}, and
helps to account for the stratigraphically late (Ordovician) appearance
of Craniids in the fossil record. (Aragonite is underrepresented in
early Palaeozoic strata due to taphonomic bias.)

The relationship of Craniiforms with respect to Linguliforms and
Rhynchonelliforms remains unclear. Shell characters point to a
relationship with the Rhynchonelliforms, which is countered by
similarities between the spermatozoa of phoronids and terebratulids,
which indicate a craniiform + linguliform clade.

It's worth noting that Bayesian and Fitch analyses place
\emph{Gasconsia} as the basalmost member of the Rhynchonellid lineage,
upholding suggestions \citep{Holmer2014OrdovicianSilurian} of a chileid
rather than trimerellid affinity. This placement presumably represents
an artefact resulting from the incorrect handling of inapplicable data.
But if true, \emph{Gasconsia} would be a close analogue for the common
ancestor of Rhynchonelliforms + Craniiforms (+Linguliforms?).
\item[Rhynchonelliforms]
The position of kutorginids within the rhynchonelliform stem lineage has
been tricky to resolve \citep{Holmer2018Theattachment}; we resolve them
as paraphyletic with respect to Rhynconellata (which encompasses the
obolellate \emph{Alisina}), which is broadly in accord to previous
proposals \citep{Holmer2018Evolutionarysignificance}. Chileids form the
adelphiotaxon to this clade. \emph{Longtancunella}
\citep{Zhang2011Theexceptionally} nests crownwards of the protorthid
\emph{Glyptoria}, but stemward of the obolellid \emph{Alisina}.

\emph{Salanygolina} has been interpreted as a stem-group
rhynchonelliform based on its combination of paterinid and chileate
features \citep{Holmer2009Theenigmatic}. Our results position
\emph{Salanygolina} between paterinids and chileids, which directly
corroborates this proposed phylogenetic position.

Basal rhynchonellids are characterized by a circular umbonal perforation
in the ventral valve, associated with a colleplax. Partly on this basis,
the aberrant taxa \emph{Yuganotheca} and \emph{Tomteluva} plot close to
\emph{Salanygolina}, the three often forming a clade -- though the
reliability of this grouping is perhaps liable to change as additional
data comes to light. Nevertheless, an interpretation of
\emph{Yuganotheca} as a stem-group brachiopod \citep{Zhang2014Anearly}
is difficult to reconcile with the increasingly well-constrained nature
of the early brachiopod body plan.
\item[Linguliforms]
The reconstruction of Linguloformea comprising Linguloidea as sister to
Discinoidea is as expected, though it is notable that Acrotretids and
Siphonotretids plot more closely to Linguloidea than Discinoidea does.

Lingulellotretids also sit within this lingulid grouping; a position in
the phoronid stem lineage \citep[advocated
by][]{Balthasar2009EarlyCambrian} is not upheld.

More novel is the reconstruction of the calcitic obolellid
\emph{Mummpikia} in the linguliform total group: a rhynchonelliform
affinity has been assumed based on its calcitic mineralogy. This said,
\citet{Balthasar2008iMummpikia} has highlighted the similarities between
obolellids and linguliform brachiopods, including sub-μm vertical canals
and the detailed configuration of the posterior shell margin. Our
analysis upholds the case for a linguliform affinity for
\emph{Mummpikia}; a calcitic shell seemingly arose through an
independent change within this taxon As such, \emph{Mummpikia} has no
direct bearing on the origin of `Calciata', save that shell mineralogy
is perhaps less static than commonly assumed.

More generally, our results identify Class Obolellata as polyphyletic:
\emph{Alisina} (Trematobolidae) plots within Rhynchonellata;
\emph{Tomteluva} is harder to place, but tends to group with
\emph{Salanygolina} stemwards of the chileids.
\item[Paterinids]
Paterinids have traditionally been placed within the Linguliforms on the
basis of their phosphatic shell \citep{Williams2007Supplement}, which
our analysis identifies as ancestral within the brachiopod crown group;
our analysis places them within the Rhynchonelliforms instead.
Characters supporting this position include the strophic hinge line,
planar cardinal area, the absence of a pedicle nerve impression, and the
morphology of the mantle canals.

More generally, although some lingulids can be found which share more
generic characters (e.g.~shell growth direction) with paterinids, the
particular combination of characters exhibited in paterinids does not
occur anywhere in the linguliform lineage, but is more similar to that
of basal rhynchonelliforms, particularly \emph{Salanygolina}.
\item[Tommotiids]
Tommotiids represent a basal grade, paraphyletic to phoronids and
crown-group brachiopods, in line with previous interpretations.

\emph{Micrina} and \emph{Mickwitzia} are the most crownwards of the
tommotiids, but beyond this, their position is somewhat difficult to pin
down; certain analytical configruations reconstruct then as
stem-brachiopods; others place them closer to the discinids, the
lingulids or the craniiforms. \emph{Heliomedusa} is commonly associated
closely with \emph{Mickwitzia}, reflecting the similarities emphasized
by Holmer and Popov in \citet{Williams2007Supplement}, but plots instead
within the Craniiforms under certain analytical conditions, in line with
earlier interpretations \citep{Williams2000LinguliformeaCraniiformea}.
\item[Hyoliths]
Hyoliths are interpreted as stem-group Brachiopods, which refines the
broader phylogenetic position proposed by
\citet{Moysiuk2017Hyolithsare}. This is to say, they sit closer to
brachiopods than the phoronids do, but no analysis places them within
the Brachiopod crown group.

Hyoliths thus represent derived tommotiids, and are the closest
relatives to the Brachiopod crown group.
\end{description}

\hypertarget{figures}{\chapter*{Supplementary Figures}\label{figures}}
\addcontentsline{toc}{chapter}{Supplementary Figures}

\begin{center}\includegraphics[width=0.8\linewidth]{images/image1} \end{center}

\textbf{Fig. S1. \emph{Pedunculotheca diania} Sun, Zhao et Zhu gen. et
sp. nov. from the Chengjiang Biota, Yunnan Province, China.} (a) NIGPAS
166601, external mould of dorsum with dorsal apex and pedicle foramen.
(b) NIGPAS 166597, preserving conical shell, operculum and internal soft
tissue, showing a compressed elliptic cross-section; backscatter
electron micrograph of boxed region shown in (c). (d) NIGPAS 166599b,
counterpart, juvenile conical shell with operculum showing two
longitudinal ventral grooves and circular larval shell. (e) NIGPAS
166602, conical shell with incomplete attachment structure. (f) NIGPAS
166598, broken shell with two ventral furrows and incomplete attachment
structure. (g) NIGPAS 166596, incomplete shell with one medial ventral
furrow and short attachment structure with coelomic cavity; detail of
boxed region shown in (h). (i) NIGPAS 166603, exterior of operculum.
Scale bars: 2mm (for a, b and e--g); 500~µm (for c, h and i).

Abbreviations: an = anus, cc = coelomic cavity, da = dorsal apex, es =
esophagus, in = intestine, mo = mouth, pe = pedicle, st = stomach.

\clearpage

\begin{center}\includegraphics[width=0.8\linewidth]{images/image2} \end{center}

\textbf{Fig. S2. Elemental distribution in the gut of
\emph{Pedunculotheca diania} Sun, Zhao et Zhu gen. et sp. nov.} NIGPAS
166597. Region corresponds to boxed region in Fig. S1c. Scale bar =
100~µm.

Abbreviations: BE = backscatter electron image, O = Oxygen, Si =
Silicon, Al = Aluminium, Fe = Iron, C = Carbon.

\clearpage

\includegraphics{Brachiopod_phylogeny_files/figure-latex/brach-diversity-1.pdf}

\textbf{Fig. S3. Global diversity of brachiopods through the Paleozoic.}
Points represent number of genera reported in each time bin; lines
represent rolling mean diversity over three consecutive time bins. Data
from Paleobiology database.

\clearpage

\hypertarget{table}{\chapter*{Supplementary Table}\label{table}}
\addcontentsline{toc}{chapter}{Supplementary Table}

\begin{tabular}{l|l|l}
\hline
NIGPAS Specimen numbers & Fossil locality & Coordinates\\
\hline
166593, 166617 & Shankou Village, Anning & 24°49’53’’ N, 102°24’47.9” E\\
\hline
166594, 166595 & Yaoying Village, Wuding & 25°36’01.2” N, 102°20’04.6” E\\
\hline
166596--166616 & Ma'anshan Village, Chengjiang & 24°40’37.2” N, 102°58’40.2” E\\
\hline
\end{tabular}

\textbf{Table S1. Provenance of fossil material.} Individuals from the
Yaoying section are usually bigger, with a thicker body wall, and have a
smaller ratio of apertural width to shell length than specimens from
other areas. In the absence of other differentiating features, we
consider these deviations to represent ecophenotypical variation within
a single species, perhaps reflecting the increased energetics and
predation pressure that accompany the shallower water depth reported at
the Yaoying section \citep{Zhao2012}.

\clearpage

\bibliography{References.bib,MorphoBank.bib}


\end{document}
